apps/src/dance/lab2/views/GenerateDancer.tsx

```diff
@@ -112,23 +115,36 @@ const GenerateDancer: React.FunctionComponent<DancerGenerateProps> = ({
           (key, index) => {
             const options = adlibsValue[adlibOption].options[key];
 
+            totalKeyCount++;
+
             if (
               options
                 .map(option => option.id)
                 .includes(lastChoices?.[index] || '')
             ) {
               // Use a value from the last saved dancer.
               initial[key] = lastChoices?.[index] || '';
+              lastKeyCount++;
```

Comment:
To make sure I'm tracking - this is checking if the loaded generated dancer has the same amount of keys as the adlib for this level? Any reason we shouldn't just compare the values directly?